### PR TITLE
Add netbsd amd64 supprot

### PR DESCRIPTION
**Pull request by posix4e**
This is dependent on https://github.com/rust-lang/rust-installer/pull/38. Once it is merged we most likely need to update the commit.

### CODE REVIEW

File: src/liblibc/lib.rs [hunk 1]
```diff
@@ -6019,7 +6024,9 @@ pub mod funcs {
     }
 
 
-    #[cfg(any(target_os = "windows",
+    //TODO SORT!
```
Comment (alexcrichton) [inline src/liblibc/lib.rs]: Could you leave this out?

File: src/liblibc/lib.rs [hunk 2]
```diff
@@ -6027,8 +6034,9 @@ pub mod funcs {
               target_os = "freebsd",
               target_os = "dragonfly",
               target_os = "bitrig",
-              target_os = "openbsd",
-              target_os = "nacl"))]
+              target_os = "nacl",
+              target_os = "netbsd",
+              target_os = "openbsd"))]
```
Comment (alexcrichton) [inline src/liblibc/lib.rs]: Android wasn't here before, so just confirming it was an intentional addition?

File: src/librustc_back/mips.rs [hunk 3]
```diff
@@ -3,8 +3,7 @@
 // http://rust-lang.org/COPYRIGHT.
 //
 // Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
-// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
-// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license // <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
```
Comment (alexcrichton) [inline src/librustc_back/mips.rs]: mishap?

File: src/libstd/sys/unix/os.rs [hunk 4]
```diff
@@ -67,7 +67,8 @@ pub fn errno() -> i32 {
         __dfly_error()
     }
 
-    #[cfg(target_os = "openbsd")]
+    // TODO Can this be combined with bitrig
```
Comment (alexcrichton) [inline src/libstd/sys/unix/os.rs]: Feel free to go ahead and combine these blocks

File: src/rust-installer [hunk 5]
```diff
@@ -1 +1 @@
-Subproject commit 8e4f8ea581502a2edc8177a040300e05ff7f91e3
+Subproject commit a25ee7f9acbc5ad477ec71f273123b71ab3d42aa
```
Comment (alexcrichton) [inline src/rust-installer]: It may be easiest to land this update separately from this PR

File: src/test/run-pass/tcp-stress.rs [hunk 6]
```diff
@@ -9,6 +9,7 @@
 // except according to those terms.
 
 // ignore-android needs extra network permissions
+// TODO This comment is weird as openbsd is not mentioned in the file
```
Comment (alexcrichton) [inline src/test/run-pass/tcp-stress.rs]: ?

File: src/liblibc/lib.rs (hunk 1 above)
Comment (posix4e) [inline src/liblibc/lib.rs]: Ah sorry I meant to merge the other commit (in the installer) first, indeed these are things that I am curious if I should do or not. There's instructions for the reviewer under here.

File: src/rust-installer (hunk 5 above)
Comment (posix4e) [inline src/rust-installer]: There's a separate pull request

File: src/test/run-pass/tcp-stress.rs (hunk 6 above)
Comment (posix4e) [inline src/test/run-pass/tcp-stress.rs]: I was making a point about the comment above it. It's not obvious to me how openbsd is relevant.

File: src/libstd/sys/unix/os.rs (hunk 4 above)
Comment (posix4e) [inline src/libstd/sys/unix/os.rs]: Thanks!

File: src/librustc_back/mips.rs (hunk 3 above)
Comment (posix4e) [inline src/librustc_back/mips.rs]: How embarrassing! I'll fix it in the commit where I remove the todos

File: src/liblibc/lib.rs (hunk 2 above)
Comment (posix4e) [inline src/liblibc/lib.rs]: I started to sort it and put that TODO sort to see if that was ok. It sounds from below that it'd be ok to go ahead and do the ordering/formatting change in the same commit as long as it is minor.

File: src/librustc_back/x86_64.rs [hunk 7]
```diff
@@ -47,7 +47,8 @@ pub fn get_target_strs(target_triple: String, target_os: abi::Os) -> target_strs
                 s0:64:64-f80:128:128-n8:16:32:64-S128".to_string()
           }
 
-          abi::OsFreebsd | abi::OsDragonfly | abi::OsBitrig | abi::OsOpenbsd => {
+	  // FIXME: Sort
+          abi::OsFreebsd | abi::OsDragonfly | abi::OsBitrig | abi::OsOpenbsd | abi::OsNetbsd => {
```
Comment (posix4e) [inline src/librustc_back/x86_64.rs]: I'm going to sort this

File: src/rust-installer (hunk 5 above)
Comment (posix4e) [inline src/rust-installer]: I'll remove this bit

File: src/test/run-pass/tcp-stress.rs (hunk 6 above)
Comment (posix4e) [inline src/test/run-pass/tcp-stress.rs]: I'm guessing the bottom line about openbsd is a lie

File: src/liblibc/lib.rs (hunk 2 above)
Comment (alexcrichton) [inline src/liblibc/lib.rs]: Yeah that's fine!

File: src/test/run-pass/tcp-stress.rs (hunk 6 above)
Comment (alexcrichton) [inline src/test/run-pass/tcp-stress.rs]: Yeah I'm not sure what's going on here either, but it's fine to just remove the TODO

File: src/test/run-pass/tcp-stress.rs [hunk 8]
```diff
@@ -9,8 +9,9 @@
 // except according to those terms.
 
 // ignore-android needs extra network permissions
-// ignore-openbsd system ulimit (Too many open files)
 // ignore-bitrig system ulimit (Too many open files)
+// ignore-netbsd system ulimit (Too many open files)
```
Comment (posix4e) [inline src/test/run-pass/tcp-stress.rs]: Curious if we should get rid of these comments. Otherwise it should be good to go

Comment (alexcrichton) [inline src/test/run-pass/tcp-stress.rs]: Eh I think it's fine for now.

**Comment by alexcrichton** [comment]
@bors: r+ 0b7c4f57f6ba59dabe4db2808fe45e8bd8bbce22 

Thanks!

**Pull request merged**